File: Demo Storyboard/LGSideMenuControllerDemo/RightViewCell.m
//
//  RightViewCell.m
//  LGSideMenuControllerDemo
//
//  Created by Grigory Lutkov on 26.04.15.
//  Copyright (c) 2015 Grigory Lutkov. All rights reserved.
//

#import "RightViewCell.h"

@implementation RightViewCell

- (void)awakeFromNib {
    [super awakeFromNib];

    // -----

    self.backgroundColor = [UIColor clearColor];

    self.textLabel.textAlignment = NSTextAlignmentCenter;
    self.textLabel.numberOfLines = 0;
    self.textLabel.lineBreakMode = NSLineBreakByWordWrapping;
}

- (void)layoutSubviews {
    [super layoutSubviews];

    self.textLabel.textColor = self.tintColor;
    _separatorView.backgroundColor = [self.tintColor colorWithAlphaComponent:0.4];
}

- (void)setHighlighted:(BOOL)highlighted animated:(BOOL)animated {
    self.textLabel.textColor = highlighted ? [UIColor colorWithRed:0.f green:0.5 blue:1.f alpha:1.f] : self.tintColor;
}

@end

File: Demo/_shared_files/SideMenuController/RightViewController/RightViewCell.swift
//
//  RightViewCell.swift
//  LGSideMenuControllerDemo
//

import Foundation
import UIKit

private let cornerRadius: CGFloat = 16.0
private let fillColorNormal: UIColor = .clear
private let fillColorHighlighted = UIColor(white: 1.0, alpha: 0.3)
private let textColorNormal: UIColor = .white
private let textColorHighlighted: UIColor = .black

class RightViewCell: UITableViewCell {

    var separatorView: UIView

    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        separatorView = UIView()

        super.init(style: style, reuseIdentifier: reuseIdentifier)

        selectionStyle = .none
        backgroundColor = .clear

        backgroundView = UIView()
        backgroundView!.backgroundColor = fillColorNormal
        backgroundView!.layer.masksToBounds = true
        backgroundView!.layer.cornerRadius = cornerRadius

        textLabel!.textAlignment = .center
        textLabel!.numberOfLines = 0
        textLabel!.lineBreakMode = .byWordWrapping
        textLabel!.textColor = textColorNormal

        separatorView.backgroundColor = UIColor.white.withAlphaComponent(0.3)
        addSubview(self.separatorView)
    }

    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    override func layoutSubviews() {
        super.layoutSubviews()

        let backgroundOffsetX: CGFloat = 8.0

        backgroundView!.frame = {
            let offsetY: CGFloat = 8.0
            return CGRect(x: backgroundOffsetX,
                          y: offsetY,
                          width: bounds.width - backgroundOffsetX + cornerRadius,
                          height: bounds.height - (offsetY * 2.0))
        }()

        textLabel!.frame = {
            let offsetX: CGFloat = 8.0
            return CGRect(x: backgroundOffsetX + offsetX,
                          y: 0.0,
                          width: bounds.width - backgroundOffsetX - (offsetX * 2.0),
                          height: bounds.height)
        }()

        separatorView.frame = CGRect(x: 16.0,
                                     y: bounds.height - 1.0,
                                     width: bounds.width - 16.0,
                                     height: 1.0)
    }

    override func setHighlighted(_ highlighted: Bool, animated: Bool) {
        super.setHighlighted(highlighted, animated: animated)
        backgroundView!.backgroundColor = highlighted ? fillColorHighlighted : fillColorNormal
        textLabel!.textColor = highlighted ? textColorHighlighted : textColorNormal
    }

}
